### PR TITLE
Pa/story 15

### PR DESCRIPTION
**Pull request by palworth**
Complete user story 15

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class SessionsController < ApplicationController
   def new
+    new_user_id
```
Comment (palworth) [inline app/controllers/sessions_controller.rb:3]: Helper method in private

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -16,6 +17,7 @@ def create
 
   def login_process(user)
     if user.default?
+      # require "pry"; binding.pry
```
Comment (palworth) [inline app/controllers/sessions_controller.rb:20]: delete this

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -26,5 +28,18 @@ def login_process(user)
     flash[:success] = "Welcome, #{user.name}!"
   end
 
-
+def new_user_id
+  if new_user = User.find_by(id: session[:user_id])
```
Comment (palworth) [inline app/controllers/sessions_controller.rb:32]: finding the user within the session. If logic to make sure that there actually is a new user

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -26,5 +28,18 @@ def login_process(user)
...
+    if new_user.admin?
+        redirect_to "/admin/dashboard"
```
Comment (palworth) [inline app/controllers/sessions_controller.rb:34]: links nested within admin because of namespacing

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -26,5 +28,18 @@ def login_process(user)
...
+        flash[:notice] = "Already logged in as #{new_user.name}"
+      elsif new_user.merchant?
```
Comment (palworth) [inline app/controllers/sessions_controller.rb:36]: new_user is a user object so we can see if it has the attribute of merchant

File: spec/features/users/login_spec.rb [hunk 6]
```diff
@@ -31,6 +31,7 @@
       expect(page).to have_content(@regular_user.email)
       expect(page).to have_content("#{@regular_user.name} is logged in.")
     end
+  end
```
Comment (palworth) [inline spec/features/users/login_spec.rb:34]: refactoring

**Review by meghanstovall — APPROVED** [review]

**Pull request merged**